src/bulls_and_cows_lib/board.cpp

```diff
@@ -1,8 +1,291 @@
 
 #include "board.hpp"
+#include "input.hpp"
+using namespace std;
 
 namespace bulls_and_cows {
 
     // TODO: define the body of the functions declared in board.cpp
 
+    string addSpaceBetweenAllChar(string s)
+    {
+        int n = (int)s.length();
+        string nouvStr = "";
+
+        for (int i = 0; i < n; i++)
```

Comment:
Au lieu d'utiliser une boucle for classique, tu peux utiliser une boucle range-for ici (https://en.cppreference.com/w/cpp/language/range-for). Ca simplifiera ton code.

Reply:
Biensur, je tâcherai à le faire, par habitude et de ce qu’on a appris c’est plutôt la boucle for classique qu’on utilise même si des fois c’est laborieux.